### PR TITLE
Fix handling of spatial reference systems

### PR DESCRIPTION
**Pull request by christophfink**
Check an input GeoDataFrame has a defined CRS, possibly reproject it to EPSG:4326 for R5, and project it back to the input CRS before returning results (if return value is GeoDataFrame).

### CODE REVIEW

**Comment by christophfink** [comment]
please **DO NOT MERGE BEFORE #138**, thanks!

**Review by wklumpen — APPROVED** [review]
Looks good to me - I see this also includes some other data validation updates.

**Comment by christophfink** [comment]
I changed the error messages of the origin/destination layer checks

File: src/r5py/util/data_validation.py [hunk 1]
```diff
@@ -22,6 +23,10 @@ def check_od_data_set(od_data_set):
...
     """
     if "id" not in od_data_set.columns:
-        raise NoIDColumnError("Origin dataset must contain an 'id' column.")
+        raise NoIDColumnError("Data set must contain an 'id' column.")
     if not od_data_set.id.is_unique:
-        raise NonUniqueIDError("Origin id values must be unique.")
+        raise NonUniqueIDError("Id values must be unique.")
+    if od_data_set.crs is None:
+        raise NoCrsError(
+            "Data set has to have a cartographic reference system defined."
```
[the remark this answers is not in the document]
Comment (christophfink) [reply src/r5py/util/data_validation.py:31]: Good catch! I had just merged it, apparently. Well, this is going to be its own issue, then.

**Pull request merged**